### PR TITLE
[wootcloud-1178] Commiting WootCloud Pack

### PR DESCRIPTION
**Pull request by lm-t**
## Status
- [] In Progress
- [ x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: None at the moment

## Description
This pull requests aims at adding the WootCloud integration to the main repo.

## Screenshots
None

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [ x] Tests
- [x ] Documentation 

## Demisto Partner?
- [ x] The title must be in the following format: **wootcloud-1178**

### CODE REVIEW

**Comment by anara123** [comment]
@lm-t Hello, thank you for your contribution!
Please fix the build, the unit tests are failing 
```
/home/circleci/project/Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py:144:14902: E231 missing whitespace after ','
 
/home/circleci/project/Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py:144:14915: E231 missing whitespace after ':'
 
/home/circleci/project/Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py:153:1: E302 expected 2 blank lines, found 1
 
/home/circleci/project/Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py:160:1: E302 expected 2 blank lines, found 1
 

 

 
#############
 
 Mypy errors 
 
#############
 

 
WootCloud
 
WootCloud.py:135:17: error: invalid syntax  [syntax]
 
        if test == :
 
                    ^
 
Found 1 error in 1 file (checked 1 source file)
 

 

 
################
 
 Vulture errors 
 
################
 

 
WootCloud
 
WootCloud.py:135: invalid syntax at "if test == :"
 

 

 
###############
 
 Pylint errors 
 
###############
 

 
WootCloud
 
************* Module WootCloud
 
WootCloud.py:135:16: E0001: invalid syntax (<unknown>, line 135) (syntax-error)
 
```

**Comment by anara123** [comment]
Also please sign the CLA 
image

File: Packs/WootCloud/Integrations/WootCloud/README.md [hunk 1]
```diff
@@ -0,0 +1,465 @@
+Append HyperContext™ insights to your SIEM data and feed them into your orchestration workflows.
+This integration was integrated and tested with version xx of WootCloud
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/README.md:2]: add `command_examples` files and generate the documentation with the provided examples
https://xsoar.pan.dev/docs/integrations/integration-docs#run-demisto-sdk-generate-docs

https://github.com/demisto/content/blob/master/Packs/HelloWorld/Integrations/HelloWorld/command_examples

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+def test_module(client):
+    """
+    Performs basic get request to get information on all WootCloud provided assets
+    """
+    # using wootassets to test API
+    test = client.http_request('GET', 'wootassets')
+    if test == :
+        pass
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:136]: this is why the build failing. fix this line.

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Host': 'api.wootuno.wootcloud.com'
+}
+# How much time to begin retrieving incidents before the first fetch
+FETCH_TIME = demisto.params().get('fetch_time', '3 days')
+# Fetch filters
+ALERT_TYPE = demisto.params().get('alert_type') # Bluetooth, Packet, Anomaly
+SEVERITY_TYPE = demisto.params().get('severity_type') # Warning, Critical, Notice, Info
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:32]: Please follow our latest code conventions https://xsoar.pan.dev/docs/integrations/code-conventions
https://xsoar.pan.dev/docs/integrations/code-conventions#main-function

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+
+''' HELPER FUNCTIONS '''
+
+def item_to_incident(item):
+    incident = {}
+    # Incident Title
+    incident['name'] = 'Incident: ' + item.get('id')
+    # Incident occurrence time, usually item creation date in service
+    incident['timestamp'] = item.get('timestamp')
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:42]:
```suggestion
    incident['occurred'] = item.get('timestamp')
```

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration test button.
+            result = test_module(client)
+            return_results('ok')
+        elif demisto.command() == 'fetch-incidents':
+            # Set and define the fetch incidents command to run after activated via integration settings.
+            fetch_incidents(client, ALERT_TYPE)
+        elif demisto.command() == 'wootcloud-get-pkt-alerts':
+            starttime, endtime = parse_date_range(demisto.args().get('date_range'), DATE_FORMAT)
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:194]: wrap each command in command function
https://xsoar.pan.dev/docs/integrations/code-conventions#command-functions

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+            alert = fetch_single_alert(client, demisto.args().get('alert_id'), 'bluetooth')
+            return_results(CommandResults(outputs=alert, outputs_prefix='WootCloud.BluetoothAlert', outputs_key_field='id'))
+        elif demisto.command() == 'wootcloud-fetch-anomaly-alert':
+            alert = fetch_single_alert(client, demisto.args().get('alert_id'), 'anomaly')
+            return_results(CommandResults(outputs=alert, outputs_prefix='WootCloud.AnomalyAlert', outputs_key_field='id'))
+    # Log exceptions
+    except Exception as e:
+        LOG(e)
+        LOG.print_log()
+        raise
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:218]:
```suggestion
        return_error('Failed to execute {} command.\nError: {}'.format(demisto.command(), str(e)))
```

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+if __name__ == "builtins":
+    main()
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:221]:
```suggestion
if __name__ in ('__main__', '__builtin__', 'builtins'):
    main()
```

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        Lists packet, bluetooth, or anomaly alerts generated in requested time span.
+        """
+        if type == "packet":
+            url = 'packetalerts'
+        elif type == "bluetooth":
+            url = 'btalerts'
+        elif type == "anomaly":
+            url = 'anomalies'
+        else:
+            return_error('Type error: %s is not one of the types' % type)
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:113]:
```suggestion
            raise ValueError('Type error: %s is not one of the types' % type)
```

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.py [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+USERNAME = demisto.params().get('client_id')
+PASSWORD = demisto.params().get('secret_key')
+SERVER = "https://api.wootuno.wootcloud.com"
+SSL_VERIFY = True
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.py:19]: make this configurable
add proxy support

File: Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+
+def test_first_fetch_incidents(requests_mock, mocker):
+    client = Client(MOCK_URL, ssl_verify=True, api_version='/v1/', headers=MOCK_HEADERS, username='test_user', password='test123')
+    requests_mock.post(MOCK_URL + '/v1/events/packetalerts', json=FETCH_ALERTS)
+
+    fetch_incidents(client, 'packet')
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud_test.py:151]: what is tested here? 
I think you should verify if fetch_incidents returns XSOAR incident properly

File: Packs/WootCloud/README.md [hunk 11]
```diff
@@ -0,0 +1 @@
+# WootCloud Pack
```
Comment (anara123) [inline Packs/WootCloud/README.md:1]: you can keep it empty

File: Packs/WootCloud/Integrations/WootCloud/WootCloud.yml [hunk 12]
```diff
@@ -0,0 +1,991 @@
...
+      description: list packet alerts generated in requested time span
+      arguments:
+      - name: date_range
+        description: Examples are (2 hours, 4 minutes, 6 month, 1 day, etc.)
+        isArray: false
+        default: true
+        required: true
+        secret: false
+      - name: severity
+        description: severity with values of 'notice', 'warning', 'critical'
```
Comment (anara123) [inline Packs/WootCloud/Integrations/WootCloud/WootCloud.yml:62]: this should be predefined/select type

**Comment by lm-t** [comment]
Closed pull request. Will look into errors and create another pull request.

**Pull request closed without merging**